aws/resource_aws_dlm_lifecycle_policy.go

```diff
@@ -0,0 +1,344 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/dlm"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsDlmLifecyclePolicy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDlmLifecyclePolicyCreate,
+		Read:   resourceAwsDlmLifecyclePolicyRead,
+		Update: resourceAwsDlmLifecyclePolicyUpdate,
+		Delete: resourceAwsDlmLifecyclePolicyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile("^[0-9A-Za-z _-]+$"), "see https://docs.aws.amazon.com/cli/latest/reference/dlm/create-lifecycle-policy.html"),
```

Comment:
As mentioned in the TODO, the docs for the API and the CLI/SDK differ. This is also the case with the `policyDetails.schedule.name` which also appears to be regex bound (not that I can even find any docs that suggest this) as well as limited to 500 characters.

I can add the validation for both `description` and `policyDetails.schedule.name` for both length and characters to give people a plan time failure rather than apply time but I don't think I can really link the AWS docs due to the inconsistent/incomplete information unfortunately :(

Reply:
yeah, understood. Is AWS aware of this kind of issue?

Reply:
I've mentioned it in my support ticket I had open with them over this and also in the linked AWS issue so hopefully they can sort that out.

Reply:
We can fix this (if necessary) after initial release since plan-time validation is generally a "nice to have" enhancement.

Reply:
👍